packages/ai-terminal/src/browser/ai-terminal-contribution.ts

```diff
@@ -0,0 +1,194 @@
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { AICommandHandlerFactory, EXPERIMENTAL_AI_CONTEXT_KEY } from '@theia/ai-core/lib/browser';
+import { CommandContribution, CommandRegistry, MenuContribution, MenuModelRegistry } from '@theia/core';
+import { KeybindingContribution, KeybindingRegistry } from '@theia/core/lib/browser';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { TerminalService } from '@theia/terminal/lib/browser/base/terminal-service';
+import { TerminalMenus } from '@theia/terminal/lib/browser/terminal-frontend-contribution';
+import { TerminalWidgetImpl } from '@theia/terminal/lib/browser/terminal-widget-impl';
+import { AiTerminalAgent } from './ai-terminal-agent';
+
+const AI_TERMINAL_COMMAND = {
+    id: 'ai-terminal:open',
+    label: 'Ask the AI'
+};
+
+@injectable()
+export class AiTerminalCommandContribution implements CommandContribution, MenuContribution, KeybindingContribution {
+
+    @inject(TerminalService)
+    protected terminalService: TerminalService;
+
+    @inject(AiTerminalAgent)
+    protected terminalAgent: AiTerminalAgent;
+
+    @inject(AICommandHandlerFactory)
+    protected commandHandlerFactory: AICommandHandlerFactory;
+
+    registerKeybindings(keybindings: KeybindingRegistry): void {
+        keybindings.registerKeybinding({
+            command: AI_TERMINAL_COMMAND.id,
+            keybinding: 'ctrlcmd+i',
+            when: `terminalFocus && ${EXPERIMENTAL_AI_CONTEXT_KEY}`
+        });
+    }
+    registerMenus(menus: MenuModelRegistry): void {
+        menus.registerMenuAction([...TerminalMenus.TERMINAL_CONTEXT_MENU, '_5'], {
+            when: EXPERIMENTAL_AI_CONTEXT_KEY,
+            commandId: AI_TERMINAL_COMMAND.id
+        });
+    }
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(AI_TERMINAL_COMMAND, this.commandHandlerFactory({
+            execute: () => {
+                if (this.terminalService.currentTerminal instanceof TerminalWidgetImpl) {
+                    new AiTerminalChatWidget(
+                        this.terminalService.currentTerminal,
+                        this.terminalAgent
+                    );
+                }
+            }
+        }));
+    }
+}
+
+class AiTerminalChatWidget {
```

Comment:
This is not a widget 🤷  We should not call it one.

Reply:
I suggest to handle this topic in the follow up https://github.com/eclipse-theia/theia/issues/14143